lib/kb_Metrics/metricsDBs.py

```diff
@@ -1,241 +1,233 @@
-		raise
-	else:
-	    pass
-	    #pprint(update_ret.raw_result)
-	    #if update_ret.upserted_id:
-		#print(update_ret.upserted_id)
-	return update_ret
+        """
+        update_narrative_records--
+        """
+        upd_op = {
+            '$currentDate': {'recordLastUpdated': True},
+            '$set': upd_data,
+            '$setOnInsert': {'access_count': 1, 'first_access': upd_data['last_saved_at']},
+            '$inc': {'access_count': 1}
+        }
+
+        # grab handle(s) to the database collection(s) targeted
+        self.mt_narrs = self.metricsDBs['metrics'][MongoMetricsDBI._MT_NARRATIVES]
+        update_ret = None
+        try:
+            # return an instance of UpdateResult(raw_result, acknowledged)
+            update_ret = self.mt_narrs.update_one(upd_filter, upd_op, upsert=True)
+        except BulkWriteError as bwe:
+            # print("mt_users.update errored\n:")
+            # pprint(bwe.details['writeErrors'])
+            panic = filter(lambda x: x['code'] != 11000, bwe.details['writeErrors'])
+            if len(panic) > 0:
+                print "really panic"
+                raise
+        else:
+            pass
+            # pprint(update_ret.raw_result)
+            # if update_ret.upserted_id:
+            # print(update_ret.upserted_id)
+        return update_ret
 
     def insert_narrative_records(self, mt_docs):
-	"""
-	Insert an iterable of narrative documents
-	"""
-	if not isinstance(mt_docs, list):
-	    raise ValueError('The variable mt_docs must be a list of mutable mapping type data.')
-	
-	# grab handle(s) to the database collection(s) targeted
+        """
+        Insert an iterable of narrative documents
+        """
+        if not isinstance(mt_docs, list):
+            raise ValueError('The variable mt_docs must be a list of mutable mapping type data.')
+
+        # grab handle(s) to the database collection(s) targeted
         self.mt_act = self.metricsDBs['metrics'][MongoMetricsDBI._MT_NARRATIVES]
-	insert_ret = None
-	try:
-	    #get an instance of InsertManyResult(inserted_ids, acknowledged)
-	    insert_ret = self.mt_act.insert_many(mt_docs, ordered=True)
-	except BulkWriteError as bwe:
-	    #skip uplicate key error (code=11000)
-	    panic = filter(lambda x: x['code'] != 11000, bwe.details['writeErrors'])
-	    if len(panic) > 0:
-		print "really panic"
-		raise
-	else:
-	    #insert_ret.inserted_ids is a list
-	    pprint('Inserted {} records.'.format(len(insert_ret.inserted_ids)))
-	return insert_ret
-
-
-    ## End functions to write to the metrics database
-
-
-    ## Begin functions to query the other dbs...
+        insert_ret = None
+        try:
+            # get an instance of InsertManyResult(inserted_ids, acknowledged)
+            insert_ret = self.mt_act.insert_many(mt_docs, ordered=True)
+        except BulkWriteError as bwe:
+            # skip uplicate key error (code=11000)
+            panic = filter(lambda x: x['code'] != 11000, bwe.details['writeErrors'])
```

Comment:
@qzzhang You should make this a function. 

Please refer to:
https://lintlyci.github.io/Flake8Rules/rules/E731.html